File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        larr = [1]
        rarr = [1]
        val = 1
        for i in range(len(nums)-1):
            val *= nums[i]
            larr.append(val)
        print(larr)
        val = 1
        for i in range(len(nums)-1, 0, -1):
            val *= nums[i]
            rarr.insert(0, val)
        print(rarr)
        for i in range(len(nums)):
            nums[i] = larr[i] * rarr[i]
        return nums

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    const res = [1]
    let val = 1

    for(let i = 0; i < nums.length-1; i++){
        val *= nums[i]
        res.push(val)
    }

    val = 1
    for(let i = nums.length-1; i > 0; i--){
        val *= nums[i]
        res[i-1] *= val
    }

    return res
};